code/modules/paperwork/clipboard.dm

```diff
@@ -8,119 +11,167 @@
 	w_class = WEIGHT_CLASS_SMALL
 	throw_speed = 3
 	throw_range = 7
-	var/obj/item/pen/haspen //The stored pen.
-	var/obj/item/paper/toppaper //The topmost piece of paper.
 	slot_flags = ITEM_SLOT_BELT
 	resistance_flags = FLAMMABLE
+	/// The stored pen
+	var/obj/item/pen/pen
+	/**
+	 * The topmost piece of paper
+	 *
+	 * Additionaly, all are in contents. This is used for the paper
+	 * displayed on the clipboard's icon and it is the one attacked,
+	 * when attacking the clipboard.
+	 * (As you can't organise contents directly in BYOND)
+	 */
+	var/obj/item/paper/toppaper
```

Comment:
Make this a weakref, since it's already inside contents and owned by the clipboard. Weakrefs will avoid the hard delete.

Reply:
I tried ~1,5 h total now, and I can't get weakrefs to work. I think I get its purpose and usage overall but probably mess some specifics up as I do not yet 100% understand everything dm.
I think it's best if someone could swap that out (probs a thing of 10 Minutes for experienced weakreffers) and I'll get those details in my head straight from the resulting code for future reference. (As I worked on this, it'll help me more than the reference I've got.)
TL;DL: Help. Weakrefs. Cannot do; need some grown-up to take my hand here 😅

Reply:
Do these docs help you out?

https://github.com/tgstation/tgstation/pull/57330/files

Reply:
Basically, this would be `var/datum/weakref/toppaper_ref`. Whenever you set it, you'd do `toppaper_ref = WEAKREF(whatever)`. Whenever you retrieve it, you'd use `var/obj/item/paper/toppaper = toppaper_ref.locate()`, then check if its null.

Reply:
![image](https://user-images.githubusercontent.com/24975989/109873884-4caf5700-7c66-11eb-801c-2df937161918.png)

Would this not be a better solution? It seems like this is the perfect use case for handling references with Exited() for paper removal and handle_atom_del for Paper deletion.